sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridEventConverter.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Core;
+using Azure.Messaging.EventGrid.Models;
+
+namespace Azure.Messaging.EventGrid
+{
+    /// <summary>
+    /// A custom converter that attributes the <see cref="EventGridEvent"/> type.
+    /// This allows System.Text.Json to serialize and deserialize EventGridEvent by default.
+    /// </summary>
+    internal class EventGridEventConverter : JsonConverter<EventGridEvent>
+    {
+        /// <summary>
+        /// Gets or sets the serializer to use for the data portion of the <see cref="EventGridEvent"/>. If not specified,
+        /// JsonObjectSerializer is used.
+        /// </summary>
+        /// <inheritdoc cref="JsonConverter{CloudEvent}.Read(ref Utf8JsonReader, Type, JsonSerializerOptions)"/>
```

Comment:
Can we do a partial inherit like this?   I've never seen that pattern before.

Reply:
oops